src/service/performance-impl.js

```diff
@@ -213,6 +214,38 @@ export class Performance {
     observer.observe({entryTypes: ['paint']});
   }
 
+  /**
+   * Reports first input delay.
+   * See https://wicg.github.io/event-timing/
+   */
+  registerEventTimingObserver_() {
+    if (!this.win.PerformanceEventTiming) {
+      return;
+    }
+    // Chrome doesn't implement the buffered flag for PerformanceObserver.
+    // That means we need to read existing entries and maintain state
+    // as to whether we have reported a value yet, since in the future it may
+    // be reported twice.
+    // https://bugs.chromium.org/p/chromium/issues/detail?id=725567
+    let recordedFirstInputDelay = false;
+    const processEntry = entry => {
+      if (recordedFirstInputDelay) {
+        return;
+      }
+      this.tick('fid', entry.processingStart - entry.startTime);
+      recordedFirstInputDelay = true;
+    };
+    const observer = new this.win.PerformanceObserver(list => {
```

Comment:
Should we try to share the instance of `PerformanceObserver` across this and the `paintTimingObserver`?

Reply:
Yeah let's do it - we'll have more perf metrics to record in the future. Done.